community/bolt/src/docs/dev/messaging.asciidoc

```diff
@@ -114,8 +114,15 @@ B1 01 8C 4D  79 43 6C 69  65 6E 74 2F  31 2E 30 A3
 ==== RESET
 
 The `RESET` message is a client message used to return the current session to a "clean" state.
+It will cause the session to `IGNORE` any message it is currently processing, as well as any message before `RESET` that had not yet begun processing.
```

Comment:
Love this feature :-)

Could you explain the semantics a little more? What does "currently processing" mean? Assuming that `RESET` is processed out of sequence, it must be one or more of these:

1. `RUN` has been received and passed to the Cypher engine but no response has been sent back to the client yet.
2. `RUN` has been received and executed and a response has been sent but no `PULL` or `DISCARD` message has been received by the server yet.
3. `RUN` has been received and processed, `PULL` has also been received but the response back to the client is incomplete (a summary message has not yet been sent).

Which of these applies determines what can be `IGNORED`. Presumably updates already made by the Cypher engine will be rolled back. As per the options above:

1. `RUN` and any subsequent `PULL` or `DISCARD` would all be `IGNORED`.
2. Only `PULL` or `DISCARD` would be `IGNORED`.
3. Any outstanding `RECORD`s not yet sent to the client would be thrown away and a terminating `IGNORED` would be sent as a summary message.

Given that `IGNORED` currently has the meaning "never even executed", is it appropriate for a response here? In (1) I could see that `PULL` or `DISCARD` should be `IGNORED` but in the other cases, perhaps a new summary message type would be appropriate (`KILLED`?) which would have a slightly different meaning: this action began but was killed and rolled back.